### PR TITLE
[Doc Improvement][API ME auth TTK]

### PR DESCRIPTION
**Pull request by v-ypalikila**
added content for API key and Entra SSO using TTK

### CODE REVIEW

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/create-api-message-extension.md](https://github.com/MicrosoftDocs/msteams-docs/blob/bd7978974a228d5e23094a436b8097ff2ab4eaaf/msteams-platform%2Fmessaging-extensions%2Fcreate-api-message-extension.md) | 95 | 21 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/caa7df3c-fd03-4a66-b99f-dbbbd020c69a) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/create-api-message-extension.md](https://github.com/MicrosoftDocs/msteams-docs/blob/b6eec9032cdfae81f9c6c6c652b6de3c931daa2e/msteams-platform%2Fmessaging-extensions%2Fcreate-api-message-extension.md) | 95 | 21 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/15833486-c6c5-4aba-8e2f-e100e48ebd23) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/create-api-message-extension.md](https://github.com/MicrosoftDocs/msteams-docs/blob/0d27d6c05cb61d502212e2cc20f3290e9a73fafd/msteams-platform%2Fmessaging-extensions%2Fcreate-api-message-extension.md) | 94 | 24 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/e04fc490-5d13-4c48-a7a5-5a7229e92826) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/create-api-message-extension.md](https://github.com/MicrosoftDocs/msteams-docs/blob/580e0da3cb1988101333e442af0796046a615712/msteams-platform%2Fmessaging-extensions%2Fcreate-api-message-extension.md) | 94 | 23 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/1086df84-f46d-4b53-8211-34b7c35249ba) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/create-api-message-extension.md](https://github.com/MicrosoftDocs/msteams-docs/blob/e9aa309a112bbbf8ae0900b52a727c482f81ec1f/msteams-platform%2Fmessaging-extensions%2Fcreate-api-message-extension.md) | 92 | 28 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/b4eda9d4-c1cc-413b-9f5e-973c96fe318c) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/create-api-message-extension.md](https://github.com/MicrosoftDocs/msteams-docs/blob/fe44c1d588d8369c8fff719737d6b378b0461e05/msteams-platform%2Fmessaging-extensions%2Fcreate-api-message-extension.md) | 93 | 28 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/ab9ffb5c-a9fe-486c-9eec-25e4ccd8d189) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/create-api-message-extension.md](https://github.com/MicrosoftDocs/msteams-docs/blob/4b46daae3110539573e2a42d4fee0658458dfc85/msteams-platform%2Fmessaging-extensions%2Fcreate-api-message-extension.md) | 95 | 23 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/d1cadb06-37ef-4c41-9e7f-e1897b4b86f9) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

File: msteams-platform/messaging-extensions/create-api-message-extension.md [hunk 1]
```diff
@@ -136,10 +136,16 @@ To build am API-based message extension using Teams Toolkit for Visual Studio Co
 
    # [New API](#tab/new-api)
```
Comment (SLdragon) [inline msteams-platform/messaging-extensions/create-api-message-extension.md:137]: There is a typo at the beginning of the doc:

**Comment by huimiu** [comment]
Hi @v-ypalikila, the folder structure of the API ME project created by TTK has been adjusted, and the relevant source code has been moved to `/src`. Would you mind checking whether it needs to be changed together in this PR?

- `repair/function.json` -> removed
- `repair/index.ts` -> `src/functions/repair.ts`
- `repairsData.json` -> `src/repairsData.json`

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/create-api-message-extension.md](https://github.com/MicrosoftDocs/msteams-docs/blob/a89b60f8fa6eeb2575584e3687c5a08189b42ed0/msteams-platform%2Fmessaging-extensions%2Fcreate-api-message-extension.md) | 95 | 26 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/7313182e-cce8-40d1-acad-a523eecda11c) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by v-ypalikila** [comment]
> ould you mind checking whether it needs to be changed together in this PR?

I've updated the table with the latest information from the readme files.

File: msteams-platform/messaging-extensions/create-api-message-extension.md (hunk 1 above)
Comment (v-ypalikila) [reply msteams-platform/messaging-extensions/create-api-message-extension.md:137]: updated.

File: msteams-platform/messaging-extensions/create-api-message-extension.md [hunk 2]
```diff
@@ -158,22 +164,62 @@ To build am API-based message extension using Teams Toolkit for Visual Studio Co
         |`appPackage/responseTemplates/repair.json`     |  A generated Adaptive Card that used to render API response.       |
         |`repairsData.json`    |  The data source for the repair API.       |
 
+   1. Based on the options selected in **step a**, follow these steps:
```
Comment (huimiu) [inline msteams-platform/messaging-extensions/create-api-message-extension.md:167]: 👆The above table also need to be updated.

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/create-api-message-extension.md](https://github.com/MicrosoftDocs/msteams-docs/blob/5419cdae7c11d51a73266ab037921d054dfb029e/msteams-platform%2Fmessaging-extensions%2Fcreate-api-message-extension.md) | 95 | 26 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/69302c3f-efa6-4aae-b194-b37527aba0e3) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/create-api-message-extension.md](https://github.com/MicrosoftDocs/msteams-docs/blob/a7ecafa11b736623ed96f5859ba21ee5074ed2f4/msteams-platform%2Fmessaging-extensions%2Fcreate-api-message-extension.md) | 95 | 25 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/4df05bad-5c0a-4752-ae9b-01d7eec9f431) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

File: msteams-platform/messaging-extensions/create-api-message-extension.md (hunk 2 above)
Comment (v-ypalikila) [reply msteams-platform/messaging-extensions/create-api-message-extension.md:169]: Ho @huimiu ,

I've updated the table respectively for Visual Studio and Visual Studio Code. 

thanks,
Yash

**Comment by v-ypalikila** [comment]
Hi @huimiu  and @SLdragon ,

I've incorporated the changes. Request you to review and approve the PR.

thanks,
Yash

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/create-api-message-extension.md](https://github.com/MicrosoftDocs/msteams-docs/blob/e0be383522335cd46e903d73963e0d6723da36d0/msteams-platform%2Fmessaging-extensions%2Fcreate-api-message-extension.md) | 95 | 25 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/7ca26492-6f9a-472a-916f-10efbf59f234) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Review by huimiu — APPROVED** [review]

**Review by SLdragon — APPROVED** [review]

**Review by v-deabira — APPROVED** [review]

**Review by surbhigupta12 — APPROVED** [review]

**Pull request merged**